example/is_partitioned_until_example.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright (c) Alexander Zaitsev <zamazan4ik@gmail.by>, 2017
+
+   Distributed under the Boost Software License, Version 1.0. (See accompanying
+   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+    For more information, see http://www.boost.org
+*/
+
+#include <vector>
+#include <functional>
+#include <iostream>
+
+#include <boost/algorithm/is_partitioned_until.hpp>
+
+
+namespace ba = boost::algorithm;
+
+bool isOdd(const int v1)
+{
+    return v1 % 2 != 0;
+}
+
+struct isOddComp
+{
+    bool operator()(const int v1) const
+    {
+        return v1 % 2 != 0;
+    }
+};
+
+
+int main ( int /*argc*/, char * /*argv*/ [] )
+{
+    std::vector<int> good({1, 2, 4});
+    std::vector<int> bad({1, 2, 3});
```

Comment:
We have identical example in 'is_partitioned'. It's example, so i think, here we can use C++11.